spec/models/concerns/digital_object_concerns/attribute_assignment/projects_spec.rb

```diff
@@ -21,14 +22,26 @@
       'other_projects' => other_projects.map { |p| { 'string_key' => p.string_key } }
     }
   end
+  let(:digital_object_data_with_bad_project_keys) do
+    {
+      'primary_project' => another_different_primary_project,
+      'other_projects' => [{ 'string_key' => nonexistent_project_string_key }]
+    }
+  end
 
-  context "#assign_projects" do
+  describe "#assign_projects" do
     it "sets the projects each time it's called" do
       digital_object.assign_projects(digital_object_data_with_projects)
       expect(digital_object.projects).to eq([different_primary_project] + other_projects)
 
       digital_object.assign_projects(digital_object_data_with_different_projects)
       expect(digital_object.projects).to eq([another_different_primary_project] + other_projects)
     end
+    context "with a non-existent other project string_key" do
+      let(:expected_error) { Hyacinth::Exceptions::NotFound }
```

Comment:
Completely optional suggestion, but maybe the `let(:expected_error)` doesn't buy you a lot here compared to just passing the error class directly to the `raise_error` method?

Reply:
Sounds good!